### PR TITLE
CSD Customization

### PR DESCRIPTION
**Pull request by shipyy**
Since #423 was done, i tought some little customization would be cool aswell
Screenshot_4

* Separated CSD into its own menu
* Added Custom Color Option (RGB)
* Added Custom Position
* Added CSD Update Rate

### CODE REVIEW

File: addons/sourcemod/scripting/surftimer/db/queries.sp [hunk 1]
```diff
@@ -39,10 +39,10 @@ char sql_insertmaptier[] = "INSERT INTO ck_maptier (mapname, tier) VALUES ('%s',
...
...kpoints` int(11) NOT NULL DEFAULT '1', `gradient` int(11) NOT NULL DEFAULT '3', `speedmode` int(11) NOT NULL DEFAULT '0', `centrespeed` int(11) NOT NULL DEFAULT '0', `centrehud` int(11) NOT NULL DEFAULT '1', teleside int(11) NOT NULL DEFAULT '0', `module1c` int(11) NOT NULL DEFAULT '1', `module2c` int(11) NOT NULL DEFAULT '2', `module3c` int(11) NOT NULL DEFAULT '3', `module4c` int(11) NOT NULL DEFAULT '4', `module5c` int(11) NOT NULL DEFAULT '5', `module6c` int(11) NOT NULL DEFAULT '6', `sidehud` int(11) NOT NULL DEFAULT '1', `module1s` int(11) NOT NULL DEFAULT '5', `module2s` int(11) NOT NULL DEFAULT '0', `module3s` int(11) NOT NULL DEFAULT '0', `module4s` int(11) NOT NULL DEFAULT '0', `module5s` int(11) NOT NULL DEFAULT '0', prestrafe int(11) NOT NULL DEFAULT '0', cpmessages int(11) NOT NULL DEFAULT '1', wrcpmessages int(11) NOT NULL DEFAULT '1', hints int(11) NOT NULL DEFAULT '1', csd_update_rate int(11) NOT NULL DEFAULT '1' , csd_pos_x float(11) NOT NULL DEFAULT '0.5' , csd_pos_y float(11) NOT NULL DEFAULT '0.3' , csd_r int(11) NOT NULL DEFAULT '255', csd_g int(11) NOT NULL DEFAULT '255', csd_b int(11) NOT NULL DEFAULT '255', PRIMARY KEY (`steamid`)) DEFAULT CHARSET=utf8mb4;";
```
Comment (Bara) [inline addons/sourcemod/scripting/surftimer/db/queries.sp:42]: the additional columns will be not compatible with exist tables, except your add a new version to the database upgrade.

**Review by Bara — CHANGES_REQUESTED** [review]

File: addons/sourcemod/scripting/surftimer/db/queries.sp (hunk 1 above)
Comment (shipyy) [reply addons/sourcemod/scripting/surftimer/db/queries.sp:42]: u mean something like this?
image

Comment (Bara) [reply addons/sourcemod/scripting/surftimer/db/queries.sp:42]: Yes, and don't forget to add the select check [here](https://github.com/surftimer/SurfTimer/blob/465c0ce3d6be2e56496f04717e6bf1b38dd223e1/addons/sourcemod/scripting/surftimer/sql.sp#L79) for a new column.

**Review by Bara — APPROVED** [review]
As I write in my previous approve comments, confirmation of a working version would be nice.

**Comment by tslashd** [comment]
Say no more, gonna try it out in a bit

**Comment by tslashd** [comment]
@shipyy when you select `R`, `G` or `B` from the menu to i guess input custom value for it nothing happens and here's the error log

> L 05/28/2022 - 13:34:08: [SM] Exception reported: Language phrase "ColorChangeValue" not found (arg 4)
> L 05/28/2022 - 13:34:08: [SM] Blaming: SurfTimer.smx
> L 05/28/2022 - 13:34:08: [SM] Call stack trace:
> L 05/28/2022 - 13:34:08: [SM]   [0] VFormat
> L 05/28/2022 - 13:34:08: [SM]   [1] Line 43, C:/Users/tbarz/Desktop/compile/addons/sourcemod/scripting/_cg/include\colorlib.inc::CPrintToChat
> L 05/28/2022 - 13:34:08: [SM]   [2] Line 2288, c:\Users\t\Desktop\Surftimer-Official-test_01\Surftimer-Official-test_01\addons\sourcemod\scripting\surftimer/commands.sp::ChangeColor
> L 05/28/2022 - 13:34:08: [SM]   [3] Line 3685, c:\Users\t\Desktop\Surftimer-Official-test_01\Surftimer-Official-test_01\addons\sourcemod\scripting\surftimer/commands.sp::CSDOptionsHandler

anything else in particular that you're unsure of so I can test it?

**Review by dPexxIAM — APPROVED** [review]

**Pull request merged**